### DISCUSSION TITLE
How to write RLS for a public bucket so that it only allows only SELECT and INSERT

### DISCUSSION THREAD

**Discussion by nivekithan** [supabase/supabase#2020] [Questions] [answered]
I am having a public bucket, lets call it `public-bucket`. I am wondering how can I write a `RLS` so that only `SELECT` and `INSERT` commands work without any authorization. 

I dont have prior experience with `postgres` or any sql languages.

Just to get the preview for my policy I typed `return true` in the editor and this is the policy `supabase` returns.

For `SELECT`
```sql
CREATE POLICY "Allow Select and insert 1eqsh30_0" ON storage.objects FOR
SELECT
  USING (return true);
```

For `INSERT`

```sql
CREATE POLICY "Allow Select and insert 1eqsh30_1" ON storage.objects FOR INSERT WITH CHECK (return true);
```

There some problems I am having with these policy

- It seems by `storage.objects` I am giving access to every bucket, whereas I wanted it to only apply for my `public-bucket`
- I dont even know  if `return true`  is valid sql syntax

Thanks for the help by the way

**Comment by sei0o** [comment, marked as answer]
Quote from [a quickstart guide](https://supabase.io/docs/guides/with-svelte#set-up-the-database-schema):

```sql
create policy "Avatar images are publicly accessible."
  on storage.objects for select
  using ( bucket_id = 'avatars' );

create policy "Anyone can upload an avatar."
  on storage.objects for insert
  with check ( bucket_id = 'avatars' );
```

You can replace `avatars` with `public-bucket`. `=` is equivalent to `==` in JavaScript. I'm not sure if `(return true)`  works, but the docs says `( true )` is enough and it looks simpler. I'm still new to Postgres & supabase and the quickstart has been a good starting point for me (try it if you haven't).

**Reply by nivekithan** [reply]
Thanks I will give it a try

**Reply by nivekithan** [reply]
It seems to work for me, thanks for the help
